src/io/network/endpoint.cpp

```diff
@@ -166,6 +165,7 @@ bool Endpoint::IsResolvableAddress(const std::string &address, uint16_t port) {
   };
   addrinfo *info = nullptr;
   auto status = getaddrinfo(address.c_str(), std::to_string(port).c_str(), &hints, &info);
+  if (status != 0) throw NetworkError(gai_strerror(status));
```

Comment:
This function didn't throw before. All usages are checking the return `bool`. Seems like it would be better to `return` false instead of throwing. 
If you throw, you would need to check that all execution paths are handling the throw correctly. First impression's that we are not try/catching and just relying on the bool.